Corona-Warn-App/src/main/java/de/rki/coronawarnapp/srs/ui/typeselection/SrsTypeSelectionFragmentViewModel.kt

```diff
@@ -1,15 +1,68 @@
 package de.rki.coronawarnapp.srs.ui.typeselection
 
+import androidx.lifecycle.asLiveData
 import dagger.assisted.AssistedFactory
 import dagger.assisted.AssistedInject
+import de.rki.coronawarnapp.srs.core.model.SrsSubmissionType
+import de.rki.coronawarnapp.srs.ui.typeselection.SrsTypeSelectionNavigationEvents.NavigateToCloseDialog
+import de.rki.coronawarnapp.srs.ui.typeselection.SrsTypeSelectionNavigationEvents.NavigateToMainScreen
+import de.rki.coronawarnapp.srs.ui.typeselection.SrsTypeSelectionNavigationEvents.NavigateToShareCheckins
 import de.rki.coronawarnapp.util.coroutine.DispatcherProvider
+import de.rki.coronawarnapp.util.ui.SingleLiveEvent
 import de.rki.coronawarnapp.util.viewmodel.CWAViewModel
 import de.rki.coronawarnapp.util.viewmodel.SimpleCWAViewModelFactory
+import kotlinx.coroutines.flow.MutableStateFlow
 
 class SrsTypeSelectionFragmentViewModel @AssistedInject constructor(
     dispatcherProvider: DispatcherProvider
 ) : CWAViewModel(dispatcherProvider) {
 
+    val navigation = SingleLiveEvent<SrsTypeSelectionNavigationEvents>()
+
+    private val srsTestTypeInternal = MutableStateFlow(SrsSubmissionType.SRS_OTHER)
+    val srsTestType = srsTestTypeInternal.asLiveData(context = dispatcherProvider.Default)
+
+    fun onCancel() {
+        navigation.postValue(NavigateToCloseDialog)
+    }
+
+    fun onNextClicked() {
+        navigation.postValue(NavigateToShareCheckins)
```

Comment:
We need to actually check if there are checkins present, like you did in the consent fragment.

Reply:
Added checkins case